src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -0,0 +1,375 @@
+        }
+    }
+
+    std::string_view DependenciesTable::TableName()
+    {
+        return s_DependenciesTable_Table_Name;
+    }
+
+    void DependenciesTable::Create(SQLite::Connection& connection)
+    {
+        using namespace SQLite::Builder;
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "createDependencyTable_v1_4");
+
+        StatementBuilder createTableBuilder;
+        createTableBuilder.CreateTable(TableName()).BeginColumns();
+        createTableBuilder.Column(IntegerPrimaryKey());
+
+        std::vector<DependenciesTableColumnInfo> dependenciesColumns =
+        {
+            { s_DependenciesTable_Manifest_Column_Name },
+            { s_DependenciesTable_MinVersion_Column_Name },
+            { s_DependenciesTable_PackageId_Column_Name }
+        };
+
+        for (const DependenciesTableColumnInfo& value : dependenciesColumns)
+        {
+            createTableBuilder.Column(ColumnBuilder(value.Name, Type::RowId).NotNull());
+        }
+
+        createTableBuilder.EndColumns();
+
+        createTableBuilder.Execute(connection);
+
+        StatementBuilder createPKIndexBuilder;
+        createPKIndexBuilder.CreateUniqueIndex(s_DependenciesTable_Index_Name).On(s_DependenciesTable_Table_Name).Columns({ s_DependenciesTable_Manifest_Column_Name, s_DependenciesTable_MinVersion_Column_Name, s_DependenciesTable_PackageId_Column_Name });
+        createPKIndexBuilder.Execute(connection);
+
+        savepoint.Commit();
+    }
+
+
+    void DependenciesTable::AddDependencies(SQLite::Connection& connection, const Manifest::Manifest& manifest, SQLite::rowid_t manifestRowId)
+    {
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, std::string{ s_DependenciesTable_Table_Name } + "add_dependencies_v1_4");
+
+        auto dependencies = GetDependencies(manifest, Manifest::DependencyType::Package);
+        if (!dependencies.size())
+        {
+            return;
+        }
+
+        InsertManifestDependencies(connection, manifestRowId, dependencies);
+
+        savepoint.Commit();
+    }
+
+    bool DependenciesTable::UpdateDependencies(SQLite::Connection& connection, const Manifest::Manifest& manifest, SQLite::rowid_t manifestRowId)
+    {
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, std::string{ s_DependenciesTable_Table_Name } + "update_dependencies_v1_4");
+
+        const auto dependencies = GetDependencies(manifest, Manifest::DependencyType::Package);
+        auto existingDependencies = GetDependenciesByManifestRowId(connection, manifestRowId);
+
+        // Get dependencies to add.
+        std::set<Manifest::Dependency> toAddDependencies;
+        std::copy_if(
+            dependencies.begin(),
+            dependencies.end(),
+            std::inserter(toAddDependencies, toAddDependencies.begin()),
+            [&](Manifest::Dependency dep)
+            {
+                auto depRow = GetPackageRowIdAndVersion(connection, dep);
```

Comment:
>auto depRow = GetPackageRowIdAndVersion(connection, dep); [](http://example.com/codeflow?start=16&length=57)

Why not just convert to this in `GetDependencies`?

Reply:
It would also mean that you have only one location that has to deal with package ids that don't exist.

Reply:
The intent for GetDependencies it to get the dependencies in a manifest and not do any form of database interaction for clarity sake. Retrieving dependencies from manifest and retrieving from db seems to be like 2 distinct operations. I see the potential for tight coupling in what is being suggested here.

Reply:
This is the lowest layer of the implementation where a manifest and database are involved and a function that is local to this file.  What I'm suggesting does not change the coupling of the data itself.  While it does indeed create coupling in the function, that is perfectly fine when it is contained to this file.

If there are other use cases to get just the dependencies from a manifest, you can always create a layered approach; where `GetAndLinkDependencies` uses `GetDependencies` to extract the set from the manifest and then links them with the database (again, creating only a single location that needs to deal with package ids that don't exist).  If there is no reason to need both right now, that minor refactoring could be left for when it is needed.